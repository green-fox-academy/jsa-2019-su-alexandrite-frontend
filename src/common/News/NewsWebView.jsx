import React from 'react';
import { WebView } from 'react-native-webview';
import { useNavigationParam } from 'react-navigation-hooks';
import { ActivityIndicator } from 'react-native';

export default () => (
  <WebView
    source={{ uri: useNavigationParam('url') }}
    renderLoading={() => <ActivityIndicator />}
  />
);
