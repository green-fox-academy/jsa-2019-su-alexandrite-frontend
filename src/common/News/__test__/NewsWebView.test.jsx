import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigationParam } from 'react-navigation-hooks';
import { shallow } from 'enzyme';
import { LayoutAnimation } from 'react-native';
import NewsWebView from '../NewsWebView';

jest.mock('react-navigation-hooks');
jest.mock('../../ProgressBar.jsx', () => 'ProgressBar');

describe('<NewsWebView />', () => {
  beforeEach(() => {
    useNavigationParam.mockReturnValue({
      uri: 'test uri',
    });
  });

  afterAll(() => {
    jest.resetModules();
  });

  it('Should render properly and match the snapshot', () => {
    const tree = renderer.create(
      <NewsWebView />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render ProgressBar and match the snapshot', () => {
    const wrapper = shallow(<NewsWebView />);
    const tree = renderer.create(
      wrapper.find('WebView').props().renderLoading(),
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render ProgressBar and match the snapshot', () => {
    const wrapper = shallow(<NewsWebView />);
    const mockConfigureNext = jest.fn();
    // https://github.com/testing-library/native-testing-library/issues/26#issuecomment-536998442
    jest.mock('react-native/Libraries/LayoutAnimation/LayoutAnimation', () => ({
      ...require.requireActual(
        'react-native/Libraries/LayoutAnimation/LayoutAnimation',
      ),
      configureNext: mockConfigureNext,
    }));
    wrapper.find('WebView').props().onLoadProgress({ nativeEvent: { progress: 0.5 } });
    expect(mockConfigureNext).toHaveBeenCalledWith(LayoutAnimation.Presets.linear);
  });
});
