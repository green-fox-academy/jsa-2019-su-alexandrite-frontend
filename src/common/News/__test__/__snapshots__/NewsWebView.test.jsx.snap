// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsWebView /> Should render ProgressBar and match the snapshot 1`] = `
<ProgressBar
  progress={0}
/>
`;

exports[`<NewsWebView /> Should render properly and match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <RNCWebView
    cacheEnabled={true}
    javaScriptEnabled={true}
    messagingEnabled={false}
    onContentProcessDidTerminate={[Function]}
    onHttpError={[Function]}
    onLoadProgress={[Function]}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingProgress={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    source={
      Object {
        "uri": Object {
          "uri": "test uri",
        },
      }
    }
    startInLoadingState={true}
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
      ]
    }
    useSharedProcessPool={true}
  />
  <ProgressBar
    progress={0}
  />
</View>
`;
